File: dataset/avatar/Python/Code/codeforces_408_B.py
from collections import defaultdict , deque
from functools import lru_cache
from heapq import heappush , heappop
from typing import Counter
from bisect import bisect_right , bisect_left
import math
hpop = heappop
hpush = heappush
def solution ( ) :
    line = input ( )
    bought_count = Counter ( line )
    line = input ( )
    made_cout = Counter ( line )
    res = 0
    for color in made_cout :
        if color not in bought_count :
            return print ( - 1 )
        res += min ( bought_count [ color ] , made_cout [ color ] )
    print ( res )
def main ( ) :
    t = 1
    for _ in range ( t ) :
        solution ( )
main ( )

File: output/deepseek_r1/avatar/algo_based_translation/python/java/successful_translation/codeforces_408_B.java

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class codeforces_408_B {
    public static void main(String[] args) {
        solution();
    }

    public static void solution() {
        Scanner scanner = new Scanner(System.in);
        String line = scanner.nextLine();
        Map<Character, Integer> boughtCount = new HashMap<>();
        for (char c : line.toCharArray()) {
            boughtCount.put(c, boughtCount.getOrDefault(c, 0) + 1);
        }

        line = scanner.nextLine();
        Map<Character, Integer> madeCout = new HashMap<>();
        for (char c : line.toCharArray()) {
            madeCout.put(c, madeCout.getOrDefault(c, 0) + 1);
        }

        int res = 0;
        for (char color : madeCout.keySet()) {
            if (!boughtCount.containsKey(color)) {
                System.out.println(-1);
                return;
            }
            res += Math.min(boughtCount.get(color), madeCout.get(color));
        }
        System.out.println(res);
    }
}

